test/functional/p2p_blocksonly.py

```diff
@@ -78,6 +78,31 @@ def blocksonly_mode_tests(self):
         self.nodes[0].disconnect_p2ps()
         self.generate(self.nodes[0], 1)
 
+        self.log.info("Check that we sustain full-relay outbound connections to blocksonly peers when not at the full oubound limit")
+        blocksonly_peer = self.nodes[0].add_outbound_p2p_connection(P2PInterface(), p2p_idx=1, txrelay=False)
+        # Trigger CheckForStaleTipAndEvictPeers, which is scheduled every 45s (EXTRA_PEER_CHECK_INTERVAL)
+        self.nodes[0].mockscheduler(46)
+        blocksonly_peer.sync_with_ping()
+        blocksonly_peer.peer_disconnect()
+
+        self.log.info("Check that we evict full-relay outbound connections to blocksonly peers at the full outbound limit")
+        # Simulate the situation of being at the max limit of full-outbound connections by setting -maxconnections to a low value
+        self.restart_node(0, ["-maxconnections=2"])
+        # Have one blocksonly peer that is not being disconnected
+        blocksonly_peer1 = self.nodes[0].add_outbound_p2p_connection(P2PInterface(), p2p_idx=1, txrelay=False)
+        blocksonly_peer2 = self.nodes[0].add_outbound_p2p_connection(P2PInterface(), p2p_idx=2, txrelay=False)
+        self.nodes[0].mockscheduler(46)
+        blocksonly_peer2.wait_for_disconnect()
+        blocksonly_peer1.sync_with_ping()
+        blocksonly_peer1.peer_disconnect()
+
+        self.log.info("Check that we don't evict full-relay outbound connections to blocksonly peers if we are blocksonly ourselves")
+        self.restart_node(0, ["-blocksonly", "-maxconnections=1"])
```

Comment:
Not sure why anyone would do `maxconnections=1` in practice, but then if their firewall allows to find only one blocksonly node (and they are not in blocksonly), the behavior might be confusing: they will be frequently disconencted from this peer.

Perhaps, this could be logged if maxconnections=1

Reply:
the idea was to log `if maxconnect is set to 1`, so at the startup (setting this argument) time.
I don't insist though.

Reply:
I'd agree with adding this warning, but yeah, this PR is not the best place for it.